libraries/ts-command-line/src/parameters/CommandLineIntegerListParameter.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { ICommandLineIntegerListDefinition } from './CommandLineDefinition';
+import { CommandLineParameterWithArgument, CommandLineParameterKind } from './BaseClasses';
+import { EnvironmentVariableParser } from './EnvironmentVariableParser';
+
+/**
+ * The data type returned by {@link CommandLineParameterProvider.defineIntegerListParameter}.
+ * @public
+ */
+export class CommandLineIntegerListParameter extends CommandLineParameterWithArgument {
+  private _values: number[] = [];
+
+  /** @internal */
+  public constructor(definition: ICommandLineIntegerListDefinition) {
+    super(definition);
+  }
+
+  /** {@inheritDoc CommandLineParameter.kind} */
+  public get kind(): CommandLineParameterKind {
+    return CommandLineParameterKind.IntegerList;
+  }
+
+  /**
+   * {@inheritDoc CommandLineParameter._setValue}
+   * @internal
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  public _setValue(data: any): void {
+    // If argparse passed us a value, confirm it is valid
+    if (data !== null && data !== undefined) {
+      if (!Array.isArray(data)) {
+        this.reportInvalidData(data);
+      }
+      for (const arrayItem of data) {
+        if (typeof arrayItem !== 'number') {
+          this.reportInvalidData(data);
+        }
+      }
+      this._values = data;
+      return;
+    }
+
+    // If an environment variable exists, attempt to parse it as a list
+    if (this.environmentVariable !== undefined) {
+      const values: string[] | undefined = EnvironmentVariableParser.parseAsList(this.environmentVariable);
+      if (values) {
+        const parsedValues: number[] = [];
+        for (const value of values) {
+          const parsed: number = parseInt(value, 10);
+          if (isNaN(parsed) || value.indexOf('.') >= 0) {
```

Comment:
This is not actually a valid test for integers.  For example, `JSON.parse("1E-1")` evaluates to `0.1`, even though the input does not contain a `.` character.

`Number.isInteger()` might be a better test.

Reply:
Looks like this logic predates your PR, though.